### ISSUE TITLE
Incorrect number of ENI and IPs for g4dn.16xlarge

### ISSUE THREAD

**Issue by walkley** [aws/amazon-vpc-cni-k8s#1083]
When I creating cluster with g4dn.16xlarge node, ipamD will be failed to create secondary ENIs and IPs due to the number of private addresses exceed limit.

## How to reproduce
1. Create cluster with only one g4dn.16xlarge node:
```
eksctl create cluster --name g4test -t g4dn.16xlarge -N 1 -m 1 -M 1 --ssh-public-key mykey
```
2. After cluster launched, get pod status, coredns pod is always in ContainerCreating
```
[ec2-user@ip-172-31-5-62 ~]$ kubectl get pod -A
NAMESPACE     NAME                      READY   STATUS              RESTARTS   AGE
kube-system   aws-node-2tvg4            1/1     Running             0          69m
kube-system   coredns-cdd78ff87-sw5zh   0/1     ContainerCreating   0          76m
kube-system   coredns-cdd78ff87-xhkx6   0/1     ContainerCreating   0          76m
kube-system   kube-proxy-fx5m4          1/1     Running             0          69m
```

## Root cause
Login to worker node and check logs of ipamD, the root cause is the limit of IPs exceed limit as it trying to allocate **49** IP addresses:
```
{"level":"debug","ts":"2020-07-12T08:56:42.962Z","caller":"ipamd/ipamd.go:758","msg":"Found ENI eni-0bce8e083e8f5962b that has less than the maximum number of IP addresses allocated: cur=0, max=49"}
{"level":"info","ts":"2020-07-12T08:56:42.962Z","caller":"ipamd/ipamd.go:762","msg":"Trying to allocate 49 IP addresses on ENI eni-0bce8e083e8f5962b"}
{"level":"error","ts":"2020-07-12T08:56:43.274Z","caller":"ipamd/ipamd.go:762","msg":"Failed to allocate a private IP addresses on ENI eni-0bce8e083e8f5962b: PrivateIpAddressLimitExceeded: Number of private addresses will exceed limit.\n\tstatus code: 400, request id: 667a19a6-48be-4436-bb52-a9710686800c"}
```

Check the code to set maximum number of IPs for each ENI, it is 50:
https://github.com/aws/amazon-vpc-cni-k8s/blob/3c20ff4137aa567ba9a76be1835683985f05aea4/pkg/awsutils/vpc_ip_resource_limit.go#L428

From documentation of [IP addresses per network interface per instance type](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html#AvailableIpPerENI), for g4dn.16xlarge, maximum ENIs is 4, maximum Private IPv4 addresses per interface is 15.

## Versions
```
$ eksctl version
0.23.0
$ kubectl get ds/aws-node -n kube-system -o=jsonpath='{$.spec.template.spec.containers[:1].image}'
602401143452.dkr.ecr.ap-northeast-1.amazonaws.com/amazon-k8s-cni:v1.6.3
```

**Comment by fawadkhaliq** [comment]
@walkley thank you for reporting the issue. This is indeed because of a mismatch of the maximum limits. Both `InstanceENIsAvailable` and `InstanceIPsAvailable ` need to be fixed for g4dn.16xlarge as per the limits here [1]

[1] https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html#AvailableIpPerENI

**Comment by mogren** [comment]
Fix merged

**Comment by zpcl616** [comment]
v1.6.4 and v1.7 also have this bug.
You can reproduce this bug follow the steps provided by @walkley.

cluster.yaml:
`---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: g4dn16xTest
  region: cn-northwest-1

managedNodeGroups:
- name: nodegroup
  desiredCapacity: 1
  instanceType: g4dn.16xlarge
  volumeSize: 100
  ssh:
    allow: true`

Result:
`➜  Workspace kubectl get pods -A
NAMESPACE     NAME                       READY   STATUS              RESTARTS   AGE
kube-system   aws-node-q6jvj             1/1     Running             0          3h11m
kube-system   coredns-6b799d5d56-g58cn   0/1     ContainerCreating   0          3h16m
kube-system   coredns-6b799d5d56-r49cd   0/1     ContainerCreating   0          3h16m
kube-system   kube-proxy-hbxm8           1/1     Running             0          3h11m`

**Comment by zpcl616** [comment]
This fix merged to v1.6.4-rc1, bug did not merge to v1.6.4.
